src/main/java/hudson/plugins/git/util/BuildDetails.java

```diff
@@ -0,0 +1,303 @@
+ * It persists Git related information for a single build, and is used
+ * at run time to build up an {@link BuildData} object.
+ */
+@ExportedBean(defaultVisibility = 999)
+public class BuildDetails implements Action, Serializable, Cloneable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The current build.
+     */
+    public Build build;
+
+    /**
+     * The name of the SCM as given by the user.
+     */
+    public String scmName;
+
+    /**
+     * The URLs that have been referenced.
+     */
+    public Set<String> remoteUrls = new HashSet<>();
+
+    /**
+     * Allow disambiguation of the action url when multiple {@link BuildDetails} actions present.
+     */
+    @CheckForNull
+    private Integer index;
+
+    public BuildDetails(Build build) {
+        this.build = build;
+    }
+
+    public BuildDetails(Build build, String scmName) {
+        this.build = build;
+        this.scmName = scmName;
+    }
+
+    public BuildDetails(Build build, String scmName, Collection<UserRemoteConfig> remoteConfigs) {
+        this.build = build;
+        this.scmName = scmName;
+        for(UserRemoteConfig c : remoteConfigs) {
+            remoteUrls.add(c.getUrl());
+        }
+    }
+
+    /**
+     * Returns the build details display name, optionally with SCM name.
+     * This string needs to be relatively short because it is
+     * displayed in a column with other short links.  If it is
+     * lengthened, it causes the other data on the page to shift
+     * right.  The page is then difficult to read.
+     *
+     * @return build details display name
+     */
+    public String getDisplayName() {
+        if (scmName != null && !scmName.isEmpty())
+            return "Git Build Details:" + scmName;
+        return "Git Build Details";
+    }
+
+    public String getIconFileName() {
+        return jenkins.model.Jenkins.RESOURCE_PATH+"/plugin/git/icons/git-32x32.png";
+    }
+
+    public String getUrlName() {
+        return index == null ? "git" : "git-"+index;
+    }
+
+    /**
+     * Sets an identifier used to disambiguate multiple {@link BuildDetails} actions attached to a {@link Run}
+     *
+     * @param index the index, indexes less than or equal to {@code 1} will be discarded.
+     */
+    public void setIndex(Integer index) {
+        this.index = index == null || index <= 1 ? null : index;
+    }
+
+    /**
+     * Gets the identifier used to disambiguate multiple {@link BuildDetails} actions attached to a {@link Run}.
+     *
+     * @return the index.
+     */
+    @CheckForNull
+    public Integer getIndex() {
+        return index;
+    }
+
+    @Restricted(NoExternalUse.class) // only used from stapler/jelly
+    @CheckForNull
+    public Run<?,?> getOwningRun() {
+        StaplerRequest req = Stapler.getCurrentRequest();
+        if (req == null) {
+            return null;
+        }
+        return req.findAncestorObject(Run.class);
```

Comment:
I think you rather want to implement `RunAction2` and keep a `private transient Run<?, ?> run` field.